sky/utils/subprocess_utils.py

```diff
@@ -43,12 +46,54 @@ def run_no_outputs(cmd, **kwargs):
                **kwargs)
 
 
-def get_parallel_threads() -> int:
-    """Returns the number of idle CPUs."""
+def _get_thread_multiplier(cloud_str: Optional[str] = None) -> int:
+    # If using Kubernetes, we use 4x the number of cores.
+    if cloud_str and cloud_str.lower() == 'kubernetes':
```

Comment:
have we tried this scaling on other clouds? I imagine the scaling constraints should be similar?

Reply:
Not tested on other clouds, but IIRC other clouds use the SSH command runner which is sensitive to higher parallelism.